src/carnot/funcs/builtins/pprof.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <sys/sysinfo.h>
+
+#include <absl/container/flat_hash_map.h>
+
+#include <string>
+
+#include "src/shared/pprof/pprof.h"
+
+#include "src/carnot/udf/registry.h"
+#include "src/carnot/udf/udf.h"
+
+namespace px {
+namespace carnot {
+namespace builtins {
+
+using px::shared::PProfProfile;
+
+class CreatePProfRowAggregate : public udf::UDA {
+ public:
+  static udf::UDADocBuilder Doc() {
+    return udf::UDADocBuilder("Convert perf profiling data to pprof format.")
+        .Details("Converts perf profiling stack traces into pprof format.")
+        .Example("df = df.agg(pprof=('stack_trace', 'count', px.pprof))")
+        .Arg("stack_trace", "Stack trace string.")
+        .Arg("count", "Count of the stack trace string.")
+        .Returns("A single row that aggregates all the stack traces and counts into pprof format.");
+  }
+
+  void Update(FunctionContext*, const StringValue stack_trace, const Int64Value count) {
+    histo_[stack_trace] += count.val;
+  }
+
+  void Merge(FunctionContext*, const CreatePProfRowAggregate& other) {
+    for (const auto& [stack_trace, count] : other.histo_) {
+      histo_[stack_trace] += count;
+    }
+  }
+
+  StringValue Serialize(FunctionContext*) {
+    static constexpr uint32_t stack_trace_period_ms = 11;
+    const uint32_t ncpus = get_nprocs_conf();
```

Comment:
Are UDAs guaranteed to execute on the instance that collected the data? I don't have a great mental model of what happens at the kelvin vs the PEM level.

Reply:
No, that is not guaranteed. Agreed that we are not doing the full right thing here, but this is the same as what we have for the pxl script for flamegraphs right now. I wonder if we should add an issue to extend the perf profile source connector to collect this info inline with stack trace collection.

Reply:
Thanks for clarifying. Creating an issue for fixing this in the future sounds good to me. Could we link that issue here with a TODO comment?

Reply:
We removed the cpu count as an input to pprof.